internal/ingest/ingest.go

```diff
@@ -825,15 +829,51 @@ func (ing *Ingester) ingestWorkerLogic(provider peer.ID) {
 			splitAtIndex = i
 			break
 		}
+		ctxIdStr := string(ai.ad.ContextID)
```

Comment:
is there a reason to key the map as `string` rather than `[]byte`?

Reply:
Map keys may be of any type that is comparable. This does not include slices, maps, and functions.